onnxruntime/core/session/inference_session.h

```diff
@@ -671,6 +677,14 @@ class InferenceSession {
   MemoryProfiler memory_profiler_;
 #endif
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
+  // Store handles to custom op libraries so that their lifetimes extend the lifetime of the session.
+  //
+  // IMPORTANT: Needs to be declared *before* the session_state_ so that the library handles are released
+  // *after* the session_state_ is destructed.
+  std::vector<std::shared_ptr<void>> custom_op_lib_handles_;
```

Comment:
**_not_** happy with this ordering requirement. This is a first attempt at making sure that custom op lib handles are deleted appropriately.

Reply:
I think we do not want to have it as a member in the session. Let's just keep it inside the SessionOptions instance which is just a few lines above.

Reply:
Created a separate PR for managing the lifetime of custom operator library handles: https://github.com/microsoft/onnxruntime/pull/13946